zwave_js_server/model/association.py

```diff
@@ -0,0 +1,76 @@
+"""Provide a model for the association."""
+from typing import Dict, List, Optional, TypedDict
+
+
+class AssociationGroupType(TypedDict):
```

Comment:
I would prefer we use a data class.
```suggestion
from dataclasses import dataclass

@dataclass
class AssociationGroupType(TypedDict):
```

And then when you have a dict, you can do

```python
return AssociationGroupType(
  maxNodes=data['maxNodes'],
  # etc
)
```

When getting data out of the dict, use `data['maxNodes']` for required, and `data.get('profile')` for optional. Also make sure to mark the types as `Optional[int]` etc.

Reply:
Should we keep the `Type` in class name or just skip it?